ui/src/components/remote_component/DynamicHookComponent.js

```diff
@@ -0,0 +1,61 @@
+import React, { useEffect, useState } from "react";
+
+import useDynamicScript from "../../hooks/useDynamicScript";
+
+// Renderless component wrapper
+const LoadDynamicScript = ({ url, setReady, setFailed }) => {
+  const { ready, failed } = useDynamicScript({
+    url: url,
+  });
+
+  useEffect(() => {
+    setReady(ready);
+    setFailed(failed);
+  }, [setReady, setFailed, ready, failed]);
+
+  return null;
+};
+
+// Dynamic Script Loading component wrapper
+export const DynamicHookComponent = ({
```

Comment:
The changes are great! My only comment for improvement is, `DynamicHookComponent` is named and placed (under `ui/src/components/remote_component`) as though it were a generic component. It's weird that it worries about the details of `url` vs `config` and talks about "Experiment Engine". Can we move this component to  `ui/src/components/experiments` and rename it to `ExperimentEngineLoaderComponent` or something like that?